File: FrontEnd/SignUp/SignUp.js
$('#signupForm').submit(function (event) {
    event.preventDefault();

    const userData = {
        firstName: $('#firstName').val(),
        lastName: $('#lastName').val(),
        email: $('#email').val(),
        password: $('#password').val(),
        role: $('#role').val()
    };

    $.ajax({
        url: 'http://localhost:8080/api/v1/auth/signup',
        type: 'POST',
        contentType: 'application/json',
        data: JSON.stringify(userData),
        success: function (resp) {
            alert('SignUp Successfully...!');
            window.location.href = '../SignIn/SignIn.html';
        },
        error: function (xhr) {
            alert('SignUp Failed : ' + xhr.responseText);
        }
    });
});

File: BackEnd/src/main/java/lk/ijse/gdse66/quantacomit/security/auth/request/SignUp.java
package lk.ijse.gdse66.quantacomit.security.auth.request;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class SignUp {
    private String firstName;
    private String lastName;
    private String email;
    private String password;
    private String role;
}